src/Illuminate/Routing/RouteCollection.php

```diff
@@ -356,4 +399,41 @@ public function count()
     {
         return count($this->getRoutes());
     }
+
+    /**
+     * Return the CompiledUrlMatcherDumper instance for the route collection.
+     *
+     * @return \Symfony\Component\Routing\Matcher\Dumper\CompiledUrlMatcherDumper
+     */
+    public function dumper()
+    {
+        return new CompiledUrlMatcherDumper($this->toSymfonyRouteCollection());
+    }
+
+    /**
+     * Convert the collection to a Symfony RouteCollection instance.
+     *
+     * @return \Symfony\Component\Routing\RouteCollection
+     */
+    public function toSymfonyRouteCollection()
+    {
+        $symfonyRoutes = new SymfonyRouteCollection();
+
+        foreach ($this->getRoutes() as $route) {
+            // If the route doesn't have a name, we'll generate one for it
+            // and re-add the route to the collection. This way we can
+            // add the route to the Symfony route collection.
+            if (! $name = $route->getName()) {
+                $route->name($name = Str::random());
+
+                $this->add($route);
```

Comment:
For the new system I needed every route to have a name so if a route doesn't has a name, I generate one for it. I don't know if this has complications for any existing app. Does anyone have any thoughts if this could affect some apps?

Reply:
Isn't there a name you can create based on the input? For example, the controller::handler? or for callbacks a callback_[index]?

Reply:
Callbacks aren't supported for route caching but I could look into the controller::handler syntax yeah 👍

Reply:
There is a risk of conflicting with a name that's already defined, even if it's a small risk.

Maybe have a clear prefix, such as "unnamed.{controller}@{method}"?

Reply:
> Maybe have a clear prefix, such as "unnamed.{controller}@{method}"?

Yeah we can definitely do that 👍

Reply:
@frankdejonge I decided to keep the randome string since it's concise and random. Otherwise you get the action twice when you do `route:list`:

<img width="1336" alt="Screenshot 2020-01-21 at 21 30 18" src="https://user-images.githubusercontent.com/594614/72840789-4b56e100-3c95-11ea-8d72-1a9574da7dd5.png">